File: Reverse Integer/Reverse Integer.java
public class Solution {
    public int reverse(int x) {
        long res = 0;
        boolean mark = res < 0? true:false;
        while (x != 0) {
            res = res * 10 + x%10;
            x /= 10;
        }
        if (mark) res = -res;
        if (res > Integer.MAX_VALUE) return Integer.MAX_VALUE;
        else if (res < Integer.MIN_VALUE) return Integer.MIN_VALUE;
        else return (int)res;
    }
}

File: Reverse Integer/Reverse Integer.cpp
class Solution {
public:
    int reverse(int x) {
        long long res = 0;
        bool mark = res < 0 ? true : false;
        while (x != 0) {
            res = res * 10 + x % 10;
            x /= 10;
        }
        if (mark) res = -res;
        if (res > INT_MAX) return INT_MAX;
        else if (res < INT_MIN) return INT_MIN;
        else return (int)res;
    }
};